api/src/main/java/org/openmrs/ConceptNumeric.java

```diff
@@ -210,14 +211,19 @@ public void setDisplayPrecision(Integer displayPrecision) {
 	}
 	
 	public Boolean getAllowDecimal() {
-		return isAllowDecimal();
+		return allowDecimal == null ? false : allowDecimal;
 	}
```

Comment:
Actually this prevents NPE when unboxing (Boolean to boolean). A null cannot be converted to a primitive boolean.